src/mito2/src/read/scan_region.rs

```diff
@@ -620,3 +687,43 @@ impl ScanInput {
         self.files.iter().map(|file| file.file_id()).collect()
     }
 }
+
+/// A partition of a scanner to read.
+/// It contains memtables and file ranges to scan.
+#[derive(Default)]
+pub(crate) struct ScanPart {
+    /// Memtables to scan.
+    /// We scan the whole memtable now. We might scan a range of the memtable in the future.
+    pub(crate) memtables: Vec<MemtableRef>,
+    /// File ranges to scan.
+    pub(crate) file_ranges: Vec<FileRange>,
+}
+
+impl fmt::Debug for ScanPart {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "ScanPart({} memtables, {} file ranges)",
+            self.memtables.len(),
+            self.file_ranges.len()
+        )
+    }
+}
+
+/// Strategy to build [ScanPart] for parallel scan.
+pub(crate) trait ScanPartBuilder {
```

Comment:
😣 why `ScanPart` is a struct but its builder is a trait

Reply:
The SeqScanner needs to create parts with a different strategy.......

Reply:
Refactor in https://github.com/GreptimeTeam/greptimedb/pull/3992/commits/78163d791b8f6c327700850f5e107ef6ed49093d

Now this trait is replaced by a collector trait that only collects file ranges. A trait is still necessary because I need to use another container in SeqScan later.